app/services/user_destroyer.rb

```diff
@@ -114,10 +114,9 @@ def destroy(user, opts = {})
       end
     end
 
-    # After the user is deleted, remove the reviewable
-    if reviewable = ReviewableUser.pending.find_by(target: user)
-      reviewable.perform(@actor, :delete_user)
-    end
+    # After the user is deleted, remove the reviewable unless request comes from reviewable
+    reviewable = ReviewableUser.pending.find_by(target: user)
+    reviewable.perform(@actor, :delete_user) if reviewable && !opts[:from_reviewable]
```

Comment:
We can be slightly more efficient here by not even looking up the reviewable if `!opts[:from_reviewable]` is not `true`.